packages/syft/src/syft/store/mongo_document_store.py

```diff
@@ -211,23 +232,43 @@ def _set(
     ) -> Result[SyftObject, str]:
         write_permission = ActionObjectWRITE(uid=obj.id, credentials=credentials)
         can_write = self.has_permission(write_permission)
+
+        store_query_key = self.settings.store_key.with_obj(obj)
+        collection_status = self.collection
+        if collection_status.is_err():
+            return collection_status
+        collection = collection_status.ok()
+
+        store_key_exists = (
+            collection.find_one({"_id": store_query_key.value}) is not None
```

Comment:
can we assume this is a always id?

Reply:
Can't be sure since `store_query_key` is a `QueryKey` and its `value` has type `Any`
![image](https://github.com/OpenMined/PySyft/assets/88959106/3496ce0b-ad80-441e-a10b-2e271a917f47)

Reply:
`store_query_key.as_dict_mongo` handles that logic. Updated the logic there.